src/Variables/MultipleChoiceVariable.php

```diff
@@ -18,40 +18,37 @@
 use Collecthor\SurveyjsParser\Values\StringValue;
 use Collecthor\SurveyjsParser\Values\SystemMissingValue;
 use Collecthor\SurveyjsParser\Values\ValueSet;
+use InvalidArgumentException;
 
 class MultipleChoiceVariable implements ClosedVariableInterface
 {
     use GetName, GetTitle, GetRawConfiguration;
 
     /**
-     * @var array<string, ValueOptionInterface>
+     * We can say this is non-empty, since valueoptions is non-empty, and this is a direct mapping from valueoptions
+     * @var non-empty-array<string, ValueOptionInterface>
      */
-    private array $valueMap = [];
+    private array $valueMap;
 
     /**
      * @param string $name
      * @param array<string, string> $titles
      * @param non-empty-list<ValueOptionInterface> $valueOptions
      * @param array<string, mixed> $rawConfiguration
+     * @param non-empty-list<string> $dataPath
      */
     public function __construct(
-        string $name,
-        array $titles,
+        private readonly string $name,
+        private readonly array $titles,
         array $valueOptions,
-        /**
-         * @phpstan-var non-empty-list<string>
-         */
-        private array $dataPath,
-        array $rawConfiguration = []
+        private readonly array $dataPath,
+        private readonly array $rawConfiguration = []
     ) {
-        $this->name = $name;
-
+        /* @phpstan-ignore-next-line */
+        assert(count($valueOptions) > 0, throw new InvalidArgumentException('ValueOptions must not be empty'));
```

Comment:
Why use PHPs `assert`?

> Assertions should be used as a debugging feature only. You may use them for sanity-checks that test for conditions that should always be true and that indicate some programming errors if not or to check for the presence of certain features like extension functions or certain system limits and features.

Reply:
This is exactly what we discussed, right? Or should just a regular if-statement be used here?

Reply:
I used the word assertions not referring to the php `assert` function. Static analysers have assertions in the phpdoc.